app/views/external_users/admin/external_users/_form.html.haml

```diff
@@ -25,24 +25,21 @@
 
     - if current_user.persona.admin?
       = f.govuk_check_boxes_fieldset :roles, legend: { text: t('.roles'), size: 's' } do
-
-        = f.govuk_check_box :roles,
-          @external_user.available_roles.first,
-          label: { text: 'Admin' },
-          link_errors: true
-
-        = f.govuk_check_box :roles,
-          @external_user.available_roles.last,
-          label: { text: 'Advocate' } do
-
-          - if current_user.persona.provider.chamber?
-            = f.govuk_collection_radio_buttons :vat_registered,
-              [['Yes','true'],['No','false']],
-              :last,
-              :first,
-              inline: true,
-              legend: { text: t('.vat_registered'), size: 's' }
-
+        - @external_user.available_roles.each_with_index do |role, idx|
+          = f.govuk_check_box(:roles,
+            role,
+            label: { text: role.humanize },
+            link_errors: idx.eql?(0)) do
+
+            - if role.eql?('advocate') && current_user.persona.provider.chamber?
+              = f.govuk_collection_radio_buttons :vat_registered,
+                [['Yes','true'],['No','false']],
+                :last,
+                :first,
+                inline: true,
+                legend: { text: t('.vat_registered'), size: 's' }
+
+      / TODO: The hint needs to be checked because an LGFS & AGFS "firm" might confuse this with an LGFS supplier number for the user. I believe the value must only be the AGFS supplier number for the firm (the default value is populated as such if it has one), or users if different. For an LGFS only firm, I believe this field is superfluos, as the users work under the supplier number of the firm alone.
```

Comment:
superfluos -> superfluous

Also (nit) can the long comment be broken over multiple lines, such as;

```suggestion
      / TODO: The hint needs to be checked because an LGFS & AGFS "firm" might confuse this
      /       with an LGFS supplier number for the user. I believe the value must only be the
      /       AGFS supplier number for the firm (the default value is populated as such if it
      /       has one), or users if different. For an LGFS only firm, I believe this field is
      /       superfluous, as the users work under the supplier number of the firm alone.
```